### DISCUSSION TITLE
Crash on opening skin editor

### DISCUSSION THREAD

**Discussion by Gamingzone123** [ppy/osu#23860] [Q&A] [answered]
***tl;dr***
i tried to edit an (apparently broken/corrupted) skin and it crashed the game, while trying to get the logs to submit a bug report here i ended up crashing the game during an update which meant that all my skins and beatmaps dissapeared. Below is logs, and the offending skin, along with steps to recreate the first error. in line with the second issue i believe it would be good to add crash handling/redundancy so this is avoided.

***full verison***
was trying to edit a skin 
my game crashed as i was trying to recreate the issue while lazer was updating and as a result it wouldn't start and i reinstalled, this caused lazer to no longer recognise the beatmaps and skins i had loaded, ~~i also lost the relevant logs...~~ i found them
~~after the update the crash no longer seems to occur, but as the patch notes didnt mention fixing this bug, im unsure if the update fixed it, or something else caused it, when the crash occured i was trying to edit the skin smoogi uses, UNTITLED or whatever it is.~~

~~Update the update didnt fix it explicitly, as the same thing happened again, but i forgot to grab the logs and am in the process of trying to recreate for logs~~

update again, i have confirmed that the issue is with the skin, it only crashes when i load this one specifically, and while yes this is a me problem, and ill use another copy of the skin going forward, this seems like a good opportunity for improvement, maybe a message saying that the skin is bad or something, rather than immediately crashing.
i will attach the logs and a copy of the problem skin below
when i exported the skin it said there were files missing locally which would not be included with export, could this be relevant?

here are the steps to recreate anyway.

make sure you have the skin selected

Steps to recreate:

1. Open osu lazer
2. navigate to beatmap selection
3. use CTRL+SHIFT+S to access skin editor
4. select gameplay
     at this point you should have a greyed out screen
5. press escape
     you will get a popup saying that an unhandled error was reported to the devs
6. press escape twice more
    you will get two more of these popups and then...
7. crash

~~additionally, does anyone know of a way to get lazer to recognise these beatmaps again aside from redownloading them all?~~
i figured it out, i duplicated the 'client_<numbers>_corrupt.realm' file and renamed the copy to 'client.realm' and this restored my beatmaps and skins.
~~i believe this is likely because the realm file has been corrupted or reset as all the hashed files are still there~~
i now believe that this is almost certainly the case due to the method of my 'fixing' it
i guess this doubles as a suggestion, add crash handling/redundancy for if the game crashes during an update being downloaded/installed

[database.log](https://github.com/ppy/osu/files/11713630/database.log)
[input.log](https://github.com/ppy/osu/files/11713631/input.log)
[legacy-ipc.log](https://github.com/ppy/osu/files/11713632/legacy-ipc.log)
[network.log](https://github.com/ppy/osu/files/11713633/network.log)
[performance.log](https://github.com/ppy/osu/files/11713634/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/11713635/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/11713636/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/11713637/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/11713638/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/11713639/runtime.log)
[updater.log](https://github.com/ppy/osu/files/11713640/updater.log)
[UNTITLED (yellow cursor_ HD) (xi9w).zip](https://github.com/ppy/osu/files/11713647/UNTITLED.yellow.cursor_.HD.xi9w.zip)

**Comment by peppy** [comment]
All I can get from this is it looks like files may be missing from your game's storage. Likely failing with null ref on line:

https://github.com/ppy/osu/blob/f8537c1cbe98bc38f0423e8eabdc2332a21d22ab/osu.Game/Database/RealmArchiveModelImporter.cs#L427

**Reply by peppy** [reply, marked as answer]
Tracking at https://github.com/ppy/osu/issues/23883
